stock_split_picking/wizards/stock_split_picking.py

```diff
@@ -28,28 +30,125 @@ def _default_picking_ids(self):
+                    continue
+                # If it's completely assigned, extract the move entirely
+                if (
+                    float_compare(
+                        move.quantity, move.product_uom_qty, precision_rounding=rounding
+                    )
+                    >= 0
+                ):
+                    moves_to_split_off += move
+                    continue
+                # Otherwise, we split the done quantities and leave the remaining
+                # quantities in the original move.
+                split_moves_vals = move.with_context(cancel_backorder=False)._split(
+                    move.product_uom._compute_quantity(
+                        move.quantity, move.product_id.uom_id, rounding_method="HALF-UP"
+                    )
+                )
+                if not split_moves_vals:
+                    continue  # pragma: no cover
+                # Adopt the move lines from the original move
+                split_moves_vals[0]["move_line_ids"] = [
+                    Command.set(move.move_line_ids.ids)
+                ]
+                new_moves_vals += split_moves_vals
+                moves_to_recompute_state += move
+            # Create the partially split off moves
+            if new_moves_vals:
+                new_moves = self.env["stock.move"].create(new_moves_vals)
+                new_moves.with_context(
+                    bypass_entire_pack=True, bypass_procurement_creation=True
+                )._action_confirm(merge=False)
+                moves_to_split_off += new_moves
+            # Recompute the state of the remaining moves
+            moves_to_recompute_state._recompute_state()
+            # If all the picking moves are the ones to be split, then it means
+            # we haven't created any backorder move. We keep the picking as-is.
+            if picking.move_ids == moves_to_split_off or not moves_to_split_off:
+                continue  # pragma: no cover
+            # Create the split orders for the extracted moves, and split them off
+            new_pickings += picking._split_off_moves(moves_to_split_off)
+        return new_pickings
 
     def _apply_move(self):
-        """Create new pickings for every move line, keep first
-        move line in original picking
+        """Apply mode `mode`: One picking per move
+
+        Extract the first move from the picking and move it to a new one.
+        Keep the rest in the original picking.
         """
         new_pickings = self.env["stock.picking"]
-        for picking in self.mapped("picking_ids"):
-            for move in picking.move_ids[1:]:
-                new_pickings += picking._split_off_moves(move)
-        return self._picking_action(new_pickings)
+        for picking in self.picking_ids:
+            if len(picking.move_ids) <= 1:
+                continue
+            new_pickings += picking._split_off_moves(picking.move_ids[0])
```

Comment:
not related to your changes but it looks like it's missing to manage cancel/done moves.
```suggestion
        for picking in self.picking_ids:
            first_move = picking.move_ids.browse()
            for move in picking.move_ids:
                if move.state in ("cancel", "done"):
                    continue
                if not first_move:
                   first_move = move
                   continue
                new_pickings += picking._split_off_moves(picking.move_ids[0])
```